examples/QIR/Simulation/SimulatorTemplate.cpp

```diff
@@ -0,0 +1,176 @@
+#include <memory>
+
+#include "QirRuntimeApi_I.hpp"
+#include "QSharpSimApi_I.hpp"
+
+namespace Microsoft
+{
+namespace Quantum
+{
```

Comment:
What version of C++ is being targeted? May be able to collapse this to `namespace Microsoft::Quantum`, depending on whether C++17 can be assumed.

Reply:
I was actually looking for this feature but didn't know it was now possible in C++ 17, thank you for pointing that out.
However, I'm now considering it might be best to retain a larger compatibility, unless the QDK/Runtime/... is already exclusively using C++ 17?

Reply:
Yes, the QIR runtime from qsharp-runtime is already using C++ 17:
https://github.com/microsoft/qsharp-runtime/blob/7bf27f0b3b1a1752c5789c9ad4c8f454b8c4e046/src/Qir/Runtime/CMakeLists.txt#L8-L10